### ISSUE TITLE
Certain ICE files trigger panic

### ISSUE THREAD

**Issue by tsunko** [HybridEidolon/rust-ages-ice-archive#1]
Files are from the PSO2 Benchmark/Char Creator. All errors point to this specific line:
https://github.com/HybridEidolon/rust-ages-ice-archive/blob/743fc9820d57f9a4f24c8c55d91d2dbe1cf7e49f/src/read.rs#L507

```
\428b9e434a9adc756743878d48d77255 
thread 'main' panicked at 'index 603109123 out of range for slice of length 4013520', read.rs:507:31
\48df543b527d05612ecee3482bf779bb 
thread 'main' panicked at 'index 3467185087 out of range for slice of length 125600', read.rs:507:31
\6c1ea932e31d02921fe0342c4eb25c72 
thread 'main' panicked at 'index 881191562 out of range for slice of length 118624', read.rs:507:31
\9217d3d256834bcaeb1a575db44662a8 
thread 'main' panicked at 'index 3599875807 out of range for slice of length 153664', read.rs:507:31
\9aa3e134becd178daa6f957907f5fad3 
thread 'main' panicked at 'index 2809670307 out of range for slice of length 107168', read.rs:507:31
\b5bcdb3756da85f975366176ea09c210 
thread 'main' panicked at 'index 1754350177 out of range for slice of length 25010816', read.rs:507:31
\b609d5ae03749638875b3795c64e4431 
thread 'main' panicked at 'index 431622033 out of range for slice of length 598976', read.rs:507:31
\b69a7ca620257a8cd375ec2cb846c456 
thread 'main' panicked at 'index 3616775084 out of range for slice of length 8096720', read.rs:507:31
\bb249483bbd959598602194c819c0077 
thread 'main' panicked at 'index 17009975 out of range for slice of length 132096', read.rs:507:31
\d727b4c40bdaa8902814c7f2d2c2c089 
thread 'main' panicked at 'index 92789336 out of range for slice of length 124960', read.rs:507:31
\e063ae9201a1e8b5079bd917d4913bad 
thread 'main' panicked at 'index 1100425273 out of range for slice of length 2113168', read.rs:507:31

**Comment by tsunko** [comment]
Further testing still needs to be done, but for `428b9e434a9adc756743878d48d77255`, the file does not need to be decrypted with the second blowfish key, PRS decompressed, nor the final XOR 95; after the first blowfish decryption, the file header info are already in plaintext.

**Comment by HybridEidolon** [comment]
@tsunko This is profoundly late, but if you'd like to try these files again, the version I just published (0.2.0) might fix this.

**Comment by tsunko** [comment]
Hi, thanks for the update - everything seems to work now! I ran the example extractor over the entirity of PSO2CC's data - ignoring the non-ICE files, the only error was the v7 file mentioned in #2.
